libecole/include/ecole/dynamics/primalsearch.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <map>
+#include <optional>
+
+#include <xtensor/xtensor.hpp>
+
+#include "ecole/dynamics/dynamics.hpp"
+
+namespace ecole::dynamics {
+
+/**
+ * An array of variable identifiers in the transformed problem.
+ */
+using VarIds = xt::xtensor<std::size_t, 1>;
+
+/**
+ * A dictionnary of variable identifiers to variable values.
+ */
+using VarIdVals = std::map<std::size_t, SCIP_Real>;
+
+class PrimalSearchDynamics : public EnvironmentDynamics<VarIdVals, std::optional<VarIds>> {
+public:
+	using ActionSet = std::optional<VarIds>;
+	using Action = VarIdVals;
+
+	int trials_per_node;
+	int depth_freq;
+	int depth_start;
+	int depth_stop;
+
+	PrimalSearchDynamics(int trials_per_node = 1, int depth_freq = 1, int depth_start = 0, int depth_stop = -1) noexcept(
+		false);
```

Comment:
Not needed (does nothing here)
```suggestion
	PrimalSearchDynamics(int trials_per_node = 1, int depth_freq = 1, int depth_start = 0, int depth_stop = -1);
```

Reply:
Done